check_db_file.py

```diff
@@ -1,27 +1,246 @@
+    #  "NEOCERA",
+    "NetShrVar",
+    "NetStreams",
+    #  "nGEM-BBTX",
+    #  "ngpspsu",
+    #  "NIMATRO",
+    #  "NWPRTXPS",
+    #  "oercone",
+    #  "PIMOT",
+    "pixelman",
+    "pr4000",
+    "procServControl",
+    "ReadASCII",
+    #  "RKNPS",
+    #  "rotating_sample_changer",
+    #  "RotBench",
+    #  "RunControl",
+    #  "RUNCTRL",
+    #  "SAMPOS",
+    "separator",
+    #  "SKFChopper",
+    #  "SKFMB350",
+    #  "SM300",
+    #  "SMC100",
+    #  "sp2xx",
+    #  "SPINFLIPPER306015",
+
+    #  "superlogics",
+    #  "TDK_LAMBDA_GENESYS",
+    #  "TEKDMM40X0",
+    #  "Tektronix_DMM_40X0",
+
+    #  "TPG",
+    #  "TRITON",
+    #  "ttiEX355P",
+    #  "ttiplp",
+    #  "TWINCAT",
+    "utilities",
+    "utilitiesApp",
+    "VisualDCT",
+    #  "wbvalve",
+    "webget",
+    #  "WinDDE",
+    #  "wm323",
+    #  "zfcntrl"
+]
+output_dir = ""
+
+
+def check_files(db_files, strict, verbose, strict_error=False):
+    total_errors = 0
+    total_strict_errors = 0
+    total_warning = 0
+    current_item = 1
+    final_item = len(db_files)
+    failed_to_parse = []
+    failed = []
+    suite = unittest.TestSuite()
     for f in db_files:
         try:
             fp = os.path.abspath(f)
-            dbc = DbChecker(fp, verbose)
-            dbc.check()
+            with open(fp) as file:
+                parsed_db = Parser(Lexer(file.read())).db()
+            suite.addTest(DbCheckerTests(parsed_db, "test_pv_check", f, verbose, strict_error))
+            if f in strict:
+                suite.addTest(DbCheckerTests(parsed_db, "test_syntax_check", f, verbose, strict_error))
+            #dbc.parse_db_file()
+            syntax_warnings = 0
+            syntax_errors = 0
+            # if f in strict:
+            #     syntax_warnings, syntax_errors = dbc.test_syntax_check()
+            # pv_warnings, pv_errors = dbc.test_pv_check()
+            # if syntax_errors > 0 or pv_errors > 0:
+            #     failed.append(f)
+            # total_warning += syntax_warnings + pv_warnings
+            # total_errors += syntax_errors + pv_errors
+            # total_strict_errors += syntax_errors
+        except DbSyntaxError as e:
+            print(e)
+            failed_to_parse.append(f)
         except IOError:
             print("FILE ERROR: File {} does not exist".format(f))
 
+    success = xmlrunner.XMLTestRunner(output=output_dir).run(suite).wasSuccessful()
+    print(output_dir)
+    if len(failed_to_parse) > 0:
+        print("Failed to parse the following files: \n{}".format(failed_to_parse))
+
 
-if __name__ == '__main__':   
+if __name__ == '__main__':
     parser = argparse.ArgumentParser()
     parser.add_argument(
         '-d', '--directory', nargs=1, default=[],
         help='The directory to search for db files in'
     )
     parser.add_argument(
-        '-f', '--files',  nargs='*', default=[],
+        '-o', '--output', nargs=1, default="",
+        help='The directory to output xml file to'
+    )
```

Comment:
If you do the following you can remove the manipulation of `args.output` below
```suggestion
        '-o', '--output', default=os.path.dirname(os.path.realpath(__file__)),
        help='The directory to output xml file to'
    )
```
The `nargs=1` makes it into a list and then you remove the list later anyway and you test if it's an empty string then give it a default, rather than just put the default in here.